### PR TITLE
macOS keyboard accessibility API alignment & support

### PR DESCRIPTION
**Pull request by HeyImChris**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

## Summary

Addresses [this issue](https://github.com/microsoft/react-native-macos/issues/702)

Align more closely with the Windows API set and add `Escape` and `Enter` support.

[Win spec](https://github.com/microsoft/react-native-windows/blob/master/vnext/proposals/active/keyboard-reconcile-desktop.md)
[Generic Msft event handler ref](https://github.com/microsoft/react-native-windows/blob/master/vnext/Microsoft.ReactNative/Views/KeyboardEventHandler.cpp#L259-L271)

TODO: Once this merges it needs to be cherry picked into the 0.62-stable and 0.63-stable branches.

## Changelog

[macOS] [Bug] - enter, escape keyboarding support

## Test Plan

Tested valid keys in RNTester

Screen Shot 2021-02-03 at 11 36 11 PM

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-macos/pull/707)

### CODE REVIEW

File: RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js [hunk 1]
```diff
@@ -54,13 +54,12 @@ class KeyEventExample extends React.Component<{}, State> {
             <View
               acceptsKeyboardFocus={true}
               enableFocusRing={true}
-              validKeysDown={['a', 'b', 'x', 'rightArrow']}
+              validKeysDown={['g', 'Esc', 'Enter', 'ArrowLeft']}
```
Comment (harrieshin) [inline RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js:57]: can we also include combination keydown input? like enter should be differently recognized from enter+ctrl

Comment (nakambo) [reply RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js:57]: +1, but I don't mind if we leave that for later

---
In reply to: [570401738](https://github.com/microsoft/react-native-macos/pull/707#discussion_r570401738) [](ancestors = 570401738)

**Review by nakambo — APPROVED** [review]
:shipit:

File: RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js (hunk 1 above)
Comment (HeyImChris) [reply RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js:57]: We don't quite have that but we do pass all modifier keys back to clients along with the key press so clients can still filter by ctrl+enter vs. enter

Comment (appden) [reply RNTester/js/examples/KeyboardEventsExample/KeyboardEventsExample.js:57]: Sorry, late to the PRty 😉 - in RNW this is represented with the `keyUpEvents` prop, which is an array of objects that do specify modifier keys as well. Also, it uses down "Escape", which is correct according to the [standard](https://www.w3.org/TR/uievents-key/#named-key-attribute-value). The [Keyboard API doc](https://github.com/microsoft/react-native-windows/blob/master/vnext/proposals/active/keyboard-reconcile-desktop.md) actually incorrectly uses "Esc", but RNW is doing the right thing. cc @harinikmsft

File: React/Views/RCTViewKeyboardEvent.m [hunk 2]
```diff
@@ -28,48 +28,48 @@ + (instancetype)keyDownEventWithReactTag:(NSNumber *)reactTag
...
                                                                  @"helpKey" : @(helpKey),
                                                              @"functionKey" : @(functionKey),
-                                                            @"leftArrowKey" : @(leftArrowKey),
-                                                           @"rightArrowKey" : @(rightArrowKey),
-                                                              @"upArrowKey" : @(upArrowKey),
-                                                            @"downArrowKey" : @(downArrowKey),
+                                                               @"ArrowLeft" : @(leftArrowKey),
+                                                              @"ArrowRight" : @(rightArrowKey),
+                                                                 @"ArrowUp" : @(upArrowKey),
+                                                               @"ArrowDown" : @(downArrowKey),
```
Comment (appden) [inline React/Views/RCTViewKeyboardEvent.m:40]: These probably shouldn't be here anymore as the `key` property now correctly represents them.

**Pull request merged**